pkg/injector/webhook.go

```diff
@@ -304,7 +304,7 @@ func (wh *webhook) mustInject(pod *corev1.Pod, namespace string) (bool, error) {
 		log.Error().Err(errNamespaceNotFound).Msgf("Error retrieving namespace %s", namespace)
 		return false, err
 	}
-	nsInjectAnnotationExists, nsInject, err := isAnnotatedForInjection(ns.Annotations)
+	nsInjectAnnotationExists, nsInject, err := isAnnotatedForInjection(ns.Annotations, ns.Kind, ns.Name)
```

Comment:
Lets just pass the entire ns object here?

Reply:
`isAnnotatedForInjection()` is used for both `Namespace` and `Pod`.
I wish we had an interface that is something like `AnnotatedNamespacedKubernetesObject`, but that does not exist.  So it seems we could keep this  OR  alternatively come up with some new type.

Reply:
Thanks, I kind of forgot that the annotations map was common for pod and namespace, this looks good!